pkg/jsonrpc2/handler.go

```diff
@@ -0,0 +1,71 @@
+package jsonrpc2
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+)
+
+type Handler struct {
+}
+
+func (*Handler) DecodeRequest(requestBytes []byte, jwtTokenFromHeader string) (Request, error) {
+	var request Request
+	err := json.Unmarshal(requestBytes, &request)
+	if err != nil {
+		return Request{}, err
+	}
+	if request.Version != JsonRpcVersion {
+		return Request{}, errors.New("incorrect jsonrpc version")
+	}
+	if request.Method == "" {
+		return Request{}, errors.New("empty method field")
+	}
+	if request.Params == nil {
+		return Request{}, ErrInvalidParams
+	}
+	if request.Auth != "" {
+		return request, nil
+	}
+
+	if jwtTokenFromHeader == "" {
+		return request, errors.New("missing auth token")
+	}
```

Comment:
I wouldn't treat this as an error. For communication between gateway nodes and DON nodes, Messages without JWT should be considered valid